File: AutomaticTimetableGeneratorSystem-main/timetable-backend/src/main/java/com/timetable/service/SectionService.java
package com.timetable.service;

import java.util.List;

import com.timetable.dto.section.SectionRequest;
import com.timetable.dto.section.SectionResponse;

public interface SectionService {

    /**
     * Save Section
     */
    SectionResponse saveSection(SectionRequest request);

    /**
     * Update Section
     */
    SectionResponse updateSection(Long sectionId,
                                  SectionRequest request);

    /**
     * Delete Section
     */
    void deleteSection(Long sectionId);

    /**
     * Get Section By ID
     */
    SectionResponse getSectionById(Long sectionId);

    /**
     * Get All Sections
     */
    List<SectionResponse> getAllSections();

    /**
     * Get Section By Code
     */
    SectionResponse getSectionByCode(String sectionCode);

    /**
     * Get Sections By Course
     */
    List<SectionResponse> getSectionsByCourse(Long courseId);

    /**
     * Get Sections By Semester
     */
    List<SectionResponse> getSectionsBySemester(Long semesterId);

}

File: AutomaticTimetableGeneratorSystem-main/timetable-frontend/src/services/sectionService.js
import api from "./api";

const BASE_URL = "/sections";

const sectionService = {

    getAllSections: async () => {

        const response = await api.get(BASE_URL);

        return response.data;
    },

    getSectionById: async (id) => {

        const response = await api.get(`${BASE_URL}/${id}`);

        return response.data;
    },

    getSectionByCode: async (code) => {

        const response = await api.get(`${BASE_URL}/code/${code}`);

        return response.data;
    },

    getSectionsByCourse: async (courseId) => {

        const response = await api.get(
            `${BASE_URL}/course/${courseId}`
        );

        return response.data;
    },

    getSectionsBySemester: async (semesterId) => {

        const response = await api.get(
            `${BASE_URL}/semester/${semesterId}`
        );

        return response.data;
    },

    createSection: async (section) => {

        const response = await api.post(
            BASE_URL,
            section
        );

        return response.data;
    },

    updateSection: async (id, section) => {

        const response = await api.put(
            `${BASE_URL}/${id}`,
            section
        );

        return response.data;
    },

    deleteSection: async (id) => {

        const response = await api.delete(
            `${BASE_URL}/${id}`
        );

        return response.data;
    }

};

export default sectionService;